### PR TITLE
fix(ext/node): add aes-128-ecb algorithm support

### LINKED ISSUE #17848: Node compatibility: "Uncaught Error: Not implemented: crypto.Cipheriv"
**Describe the bug**

Trying to use https://github.com/codetheweb/tuyapi

Attempting to connect to a device using protocol v3.3 triggers this stack-trace

    at notImplemented (https://deno.land/std@0.170.0/node/_utils.ts:23:9)
    at new Cipheriv (https://deno.land/std@0.170.0/node/internal/crypto/cipher.ts:140:5)
    at Object.createCipheriv (https://deno.land/std@0.170.0/node/crypto.ts:187:10)
    at F._encryptPre34 (https://esm.sh/v102/tuyapi@7.5.1/deno/tuyapi.js:2:1741)
    at F.encrypt (https://esm.sh/v102/tuyapi@7.5.1/deno/tuyapi.js:2:1699)
    at C._encodePre34 (https://esm.sh/v102/tuyapi@7.5.1/deno/tuyapi.js:2:8787)
    at C.encode (https://esm.sh/v102/tuyapi@7.5.1/deno/tuyapi.js:2:8678)
    at v.get (https://esm.sh/v102/tuyapi@7.5.1/deno/tuyapi.js:2:11788)
    at v._finishConnect (https://esm.sh/v102/tuyapi@7.5.1/deno/tuyapi.js:2:14825)
    at Socket.<anonymous> (https://esm.sh/v102/tuyapi@7.5.1/deno/tuyapi.js:2:16523)

**Steps to Reproduce**

Bit awkward as you need a v3.3 physical "smart" device (e.g. a smart lightbulb).  This is reliably triggered using the "basic usage" example in https://github.com/codetheweb/tuyapi

**Expected behavior**

crypto.Cipheriv implemented

**Environment**

$ deno --version
deno 1.29.3 (release, aarch64-unknown-linux-gnu)
v8 10.9.194.5
typescript 4.9.4

### PR DESCRIPTION
**Pull request by kt3k**
This PR adds `aes-128-ecb` algorithm support. (The algorithm is used by `npm:tuyapi`)

closes #17848

### CODE REVIEW

File: ext/node/polyfills/internal/crypto/cipher.ts [hunk 1]
```diff
@@ -116,6 +118,10 @@ export interface DecipherOCB extends Decipher {
   ): this;
 }
 
+function toU8(input: string | Uint8Array): Uint8Array {
+  return typeof input === "string" ? encoder.encode(input) : input;
```
Comment (littledivy) [inline ext/node/polyfills/internal/crypto/cipher.ts:122]: nit: use `core.encode` instead. Ideally we could have seperate fast ops for string and buffer args.

**Review by littledivy — APPROVED** [review]
LGTM

File: ext/node/crypto/cipher.rs [hunk 2]
```diff
@@ -156,6 +173,12 @@ impl Decipher {
           decryptor.decrypt_block_b2b_mut(input.into(), output.into());
         }
       }
+      Aes128Ecb(decryptor) => {
+        assert!(input.len() % 16 == 0);
```
Comment (littledivy) [inline ext/node/crypto/cipher.rs:177]: Is it possible to hit this assertions from userland?

Comment (kt3k) [reply ext/node/crypto/cipher.rs:177]: If the user directly called `ops.op_node_cipheriv_encrypt` with arbitrary length input, then this assert can fail. If the user use cipher only via Node.js compat APIs, this shouldn't fail.

**Pull request merged**